code/modules/power/cell.dm

```diff
@@ -34,12 +34,11 @@
 
 	onMaterialChanged()
 		..()
+		genrate = 0
+		var/rate1 = round((material.getProperty("n_radioactive") / 6.333))
+		var/rate2 = round((material.getProperty("n_radioactive") / 4.333))
+		genrate = (rate1 + rate2)
 		if (istype(src.material))
```

Comment:
```suggestion
		if (istype(src.material))
			genrate = 0
			if(src.material.hasProperty("radioactive"))
				genrate += round((material.getProperty("radioactive") / 6.33))
			if(src.material.hasProperty("n_radioactive"))
				genrate += round((material.getProperty("n_radioactive") / 4.33))
```

This is what I had in mind.
Also, I believe you accidentally used n_radioactive twice in your change, instead of radioactive?
And I think putting the calculation outside of the istype(src.material) may result in runtimes, if it has no material.

Reply:
ah